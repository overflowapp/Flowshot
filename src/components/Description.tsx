import * as React from 'react';
interface Props {
    title: string;
    titleClass?: string;
    content: string;
}

const Description = ({ title, titleClass, content }: Props) => (
    <div className="ph6 pt2 text-left">
        <h2 className={titleClass}>{title}</h2>
        <p className="mb6">{content}</p>
    </div>
);

export default Description;
