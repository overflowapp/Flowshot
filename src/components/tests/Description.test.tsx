import * as React from 'react';
import { shallow } from 'enzyme';
import Description from '../Description';

describe('Description', function() {
    const props = {
        title: 'Begin session',
        titleClass: 'begin',
        content: 'Start off your session',
    };
    const MountedDescription = shallow(<Description {...props} />);

    it('should render heading', function() {
        expect(MountedDescription.find('h2').exists()).toBe(true);
        expect(MountedDescription.find('h2').hasClass(props.titleClass)).toBe(true);
        expect(MountedDescription.find('h2').text()).toBe(props.title);
    });

    it('should render text', function() {
        expect(MountedDescription.find('p').exists()).toBe(true);
        expect(MountedDescription.find('p').text()).toBe(props.content);
    });

    it('pass snapshot', function() {
        expect(MountedDescription).toMatchSnapshot();
    });
});
