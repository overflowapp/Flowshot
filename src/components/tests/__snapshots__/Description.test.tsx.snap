// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description pass snapshot 1`] = `
<div
  className="ph6 pt2 text-left"
>
  <h2
    className="begin"
  >
    Begin session
  </h2>
  <p
    className="mb6"
  >
    Start off your session
  </p>
</div>
`;
